wnfs/src/private/directory.rs

```diff
@@ -2170,3 +2170,57 @@ mod tests {
         Ok(())
     }
 }
+
+#[cfg(test)]
+mod snapshot_tests {
+    use super::*;
+    use crate::{private::forest::hamt::HamtForest, utils};
+    use fake::{faker::chrono::en::DateTime, Fake};
+    use rand_chacha::ChaCha12Rng;
+    use rand_core::SeedableRng;
+    use wnfs_common::utils::MockStore;
+
+    #[async_std::test]
+    async fn test_private_fs() -> Result<()> {
+        let rng = &mut ChaCha12Rng::seed_from_u64(0);
+        let store = &mut MockStore::default();
+        let forest = &mut Rc::new(HamtForest::new_rsa_2048(rng));
+        let base_name = forest.empty_name();
+        let paths = [
+            vec!["text.txt".into()],
+            vec!["music".into(), "jazz".into()],
+            vec!["videos".into(), "movies".into(), "anime".into()],
+        ];
+
+        let root_dir = &mut Rc::new(PrivateDirectory::new(
+            &base_name,
+            DateTime().fake_with_rng(rng),
```

Comment:
Very minor nit: This results in a bunch of negative times. I don't know if we want to specifically support negative timestamps forever and we also don't really need to generate random ones here. What do you think of just a default time everywhere? Jan 1st 1970 or some other fun date.